app/views/course/assessment/answer/programming/_programming.json.jbuilder

```diff
@@ -47,7 +47,7 @@ json.testCases do
           json.expression test_case.expression
           json.expected test_case.expected
           if test_result
-            json.output get_output(test_result) if can_grade
+            json.output get_output(test_result) if can_grade || current_course.enable_public_test_cases_output || current_course_user.staff?
```

Comment:
Make sure that only public test cases results are sent to the frontend. 

And any reasons why you want to check the `current_course_user.staff?` here?  It should be duplicated with `can_grade`.

Reply:
I wasn't sure if can_grade will include the current_course_user.staff?, that is why I had included that extra check. As you say that the can_grade already includes the check for staff, I will remove that extra check.